File: 백준/Gold/10504. 덧셈/덧셈.cc
#include <iostream>
using namespace std;
int main(void) {
    int l, n, t, x, rem;
    string st;
    cin >> l;
    for (int i = 0; i < l; i++) {
        cin >> n;
        x = 2;
        while (1) {
            t = n - x * (x + 1) / 2;
            if (t < 0) {
                st = "";
                break;
            }
            if (t % x == 0) {
                rem = t / x;
                st = to_string(n) + " = ";
                for (int i = 1; i < x; i++)
                    st += to_string(i + rem) + " + ";
                st += to_string(x + rem);
                break;
            }
            x += 1;
        }
        cout << (st.size() == 0 ? "IMPOSSIBLE" : st) << "\n";
    }
    return 0;
}

File: 백준/Gold/10504. 덧셈/덧셈.py
import sys
input=lambda:sys.stdin.readline().rstrip()
for _ in range(int(input())):
    n=int(input())
    arr,x=[],2
    while 1:
        t=n-x*(x+1)//2
        if t<0:
            break
        if t%x==0:
            rem=t//x
            for i in range(1,x+1):
                arr+=[i+rem]
            break
        x+=1
    if len(arr):
        st=str(n)+" = ";
        for i in range(len(arr)-1):
            st+=str(arr[i])+" + "
        st+=str(arr[len(arr)-1])
        print(st)
    else:
        print("IMPOSSIBLE")